proto/orc_proto.proto

```diff
@@ -221,15 +227,29 @@ message PostScript {
   //   [0, 12] = Hive 0.12
   repeated uint32 version = 4 [packed = true];
   optional uint64 metadataLength = 5;
-  // Version of the writer:
-  //   0 (or missing) = original
+
+  // The version of the writer that wrote the file. This number is
+  // updated when we make fixes or large changes to the writer so that
+  // readers can detect whether a given bug is present in the data.
+  // These numbers are assigned from 0 per a writer.
+  //
+  // Version of the ORC Java writer:
+  //   0 = original
   //   1 = HIVE-8732 fixed
   //   2 = HIVE-4243 fixed
   //   3 = HIVE-12055 fixed
   //   4 = HIVE-13083 fixed
   //   5 = ORC-101 fixed
   //   6 = ORC-135 fixed
+  //
+  // Version of the ORC C++ writer:
+  //   0 = original
```

Comment:
In the Java, I made the test for "includes" treat the other writers as always having the bug fixed. So, PRESTO_ORIGINAL will automatically pass all of the checks looking for specific ORC or Hive jiras (HIVE-4243 or ORC-101).

That does mean that if all three implementations are broken in the same way, we'll need a test that looks like:
    if (version.includes(WriterVersion.ORC-999) && version.includes(WriterVersion.ORC-1000)
        && version.includes(WriterVersion.PRESTO-1001) {
        ... some future bug is fixed
    }